.travis.yml

```diff
@@ -4,14 +4,19 @@ language: go
 git:
   depth: 1
 
+services:
+  - docker
 before_script:
-  - CHANGED_FILES=$(git diff --name-only HEAD $(git merge-base HEAD $TRAVIS_BRANCH))
-
+  - CHANGED_FILES_PR=$(git diff --name-only HEAD $(git merge-base HEAD $TRAVIS_BRANCH))
 script:
   - make ci e2e status=keep
-
 after_success:
-  - if [[ "${CHANGED_FILES[@]}" =~ "scripts/kind-e2e/e2e.sh" ]]; then
-      echo "scripts/kind-e2e/e2e.sh was modified, testing again on already deployed environment.";
+  - if [[ "${CHANGED_FILES_PR[@]}" =~ "scripts/kind-e2e/e2e.sh" ]]; then
+      echo "scripts/kind-e2e/e2e.sh was modified, testing recurring run on already deployed infrastructure.";
       make e2e status=keep;
-    fi
\ No newline at end of file
+    fi
+deploy:
+  provider: script
+  script: bash scripts/kind-e2e/deploy.sh
```

Comment:
more than "deploy.sh" wouldn't be this "publish.sh" ? 

It's the deploy action of travis, but we are actually publishing?

And may be it should not go inside scripts/kind-e2e/  may be tools/ release-tools/ something like that, I don't know, but it's not e2e :)

Reply:
There already is a _scripts/release_ script - all it does currently is execute the _ci_ script  - not sure what it's used for (if at all) - perhaps this would be a logical place to put the deploy code.

Reply:
Yes makes sense.

Reply:
Added the push flow to the scripts/release script and tested on my repo the images are pushed after push to master.